src/App.tsx

```diff
@@ -1,14 +1,34 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
 
   render() {
+    const { query } = this.state;
+    const visibleMovies = moviesFromServer.filter((movie) => {
+      const movieTitle = movie.title.toLowerCase();
+      const movieDescription = movie.description.toLowerCase();
+
+      return (
+        movieTitle.includes(query.toLowerCase())
+        || movieDescription.includes(query.toLowerCase())
+      );
+    });
```

Comment:
Take this logic to a separate method and call it in the render method. For example: getVisibleMovies().

Reply:
Yes, good advice.